app/client/ui/ToggleEnterprisePage.ts

```diff
@@ -0,0 +1,88 @@
+import {makeT} from 'app/client/lib/localization';
+import {Computed, Disposable, dom, makeTestId} from "grainjs";
+import {AppModel} from "app/client/models/AppModel";
+import {loadingSpinner} from 'app/client/ui2018/loaders';
+import {commonUrls} from "app/common/gristUrls";
+import {cssLink} from 'app/client/ui2018/links';
+import {ToggleEnterpriseModel} from 'app/client/models/ToggleEnterpriseModel';
+import {
+  cssOptInButton,
+  cssOptInOutMessage,
+  cssOptOutButton,
+  cssParagraph,
+  cssSection,
+  cssSpinnerBox
+} from 'app/client/ui/AdminTogglesCss';
+
+
+const t = makeT('ToggleEnterprsiePage');
+const testId = makeTestId('test-toggle-enterprise-page-');
+
+export class ToggleEnterprisePage extends Disposable {
+  private readonly _model: ToggleEnterpriseModel = new ToggleEnterpriseModel(this._appModel);
+  private readonly _isEnterprise = Computed.create(this, this._model.edition, (_use, edition) => {
+    return edition === 'enterprise';
+  })
+    .onWrite(async (enabled) => {
+      await this._model.updateEnterpriseToggle(enabled ? 'enterprise' : 'core');
+    });
+   constructor(private _appModel: AppModel) {
+     super();
+     this._model.fetchEnterpriseToggle().catch(reportError);
+   }
+
+  public getEnterpriseToggleObservable() {
+    return this._isEnterprise;
+  }
+
+  public buildEnterpriseSection() {
+    return cssSection(
+      dom.domComputed(this._isEnterprise, isEnterprise => {
```

Comment:
Are you sure it can be null? The funciton returns only boolean.

Reply:
Oops, more overzealous cargo-culting.